### ISSUE TITLE
Improv provisioning stalls

### ISSUE THREAD

**Issue by thenewwazoo** [ratgdo/homekit-ratgdo#15]
During Improv-based provisioning, the various steps never "recover", and progressing requires closing the flasher UI entirely and re-connecting. Once the user has re-connected, the next step is available and succeeds (but never "recovers").

That is, instead of
* connect
* flash image
* connect wifi
* open device page

Instead a user must
* connect
* flash image
* disconnect and reconnect
* connect wifi
* disconnect and reconnect
* etc

## post-flashing

After flashing, the console log reads:
```
Wrote 542416 bytes (378008 compressed) at 0x0 in 37.034 seconds.
Leaving...
HARD RESET
DISCONNECT
Initializing Improv Serial
Starting read loop
Writing to stream: [0x49, 0x4D, 0x50, 0x52, 0x4F, 0x56, 0x01, 0x03, 0x02, 0x02, 0x00, 0xE5, 0x0A]
Finished read loop
Improv initialization failed. Error: Improv Wi-Fi Serial not detected
```

The time between `DISCONNECT` and `Initializing Improv Serial` is far, far shorter than the time required for the Improv loop to have begun running.

## post wifi

After provisioning WiFi, the console log reads:
```
Writing to stream: [0x49, 0x4D, 0x50, 0x52, 0x4F, 0x56, 0x01, 0x03, 0x1B, 0x01, 0x19, 0x07, (ssid bytes), 0x10, (psk bytes), 0x12, 0x0A]
PROCESS {version: 1, packetType: 1, packetLength: 1, data: '[0x03]'}
Uncaught (in promise) TIMEOUT
Finished read loop
```

I suspect this is because the connect_wifi loop bails after 2 seconds (10 times 200ms delay). Increase this and see what happens.

**Comment by kharamel** [comment]
Was hoping go get started with this but looks like this has been increased to 10s. Did that fix the issue?

**Comment by jgstroud** [comment]
I wonder if this is caused by https://github.com/ratgdo/homekit-ratgdo/pull/41

**Comment by jgstroud** [comment]
Fixed in #129
